app/poro/tracking_id_generator.rb

```diff
@@ -0,0 +1,36 @@
+class TrackingIdGenerator
+  attr_reader :base_date
+  MAX_TRACKING_ID_LENGTH = 4
+  INITIAL_VALUE = '0001'.freeze
+
+  def start
+    return INITIAL_VALUE unless last_tracking_id.present? && valid_moment_to_start?
+
+    formatted_value
+  end
+
+  def valid_moment_to_start?
+    Time.at(last_order.created_at).today? || !start_of_month?
```

Comment:
explicame esto, para mi solo debe iniciar el contador si es principio de mes, por que estas haciendo este metodo para que sea la negacion de eso "o " otra condicion?

Reply:
hay un caso mas, fijate en los test a ver si eso te ayuda un poco mas pero es tipo: primero verifico si la ultima orden es de hoy porque entonces si es principio de mes y la ultima orden es de hoy quiere decir que ya hubo un pedido y por lo tanto no debo reiniciar si no sumar, se entiende? los primeros dias del mes estará pasando que la condicion de si es el primer dia dará true y querrá reiniciar el contador, lo cual no es correcto si ya fue reiniciado

Reply:
got it, entendido, lo que no me queda clar es porque estas haciendo la negacion de `start_of_month?`

Reply:
por lo que te decia de que hay un caso mas: siempre que la ultima orden sea del dia hoy ya retorna con `true` pero que pasa cuando es false?: si la ultima orden NO es del dia de hoy pero resulta que NO es principio de mes entonces debo hacer la sumatoria, en caso contrario si resulta que la orden NO es del dia de hoy pero SI es principio de mes entonces porque hoy no he generado la primera orden por lo que no debo dejar que sume uno si no que reinicie el valor desde '0001'